### ISSUE TITLE
Spyder 5.0.0 start error with spyder-notebook installed

### ISSUE THREAD

**Issue by WANGCHIENCHIH** [spyder-ide/spyder#15280]
### What steps will reproduce the problem?

install spyder 5.0.0 and spyder-notebook plugin

and dependency error 
spyder_kernel need above 2.1
File "zmq/backend/cython/socket.pyx", line 542, in zmq.backend.cython.socket.Socket.bind
File "zmq/backend/cython/checkrc.pxd", line 28, in zmq.backend.cython.checkrc._check_rc
zmq.error.ZMQError: Address in use

### Traceback
```python-traceback
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\application\container.py", line 286, in report_missing_dependencies
    QMessageBox.critical(
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 27, in poll
    super().poll()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\restarter.py", line 116, in poll
    self.kernel_manager.restart_kernel(now=True, newports=newports)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 510, in restart_kernel
    self.start_kernel(**self._launch_args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 342, in start_kernel
    self.post_start_kernel(**kw)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 62, in post_start_kernel
    self.kernel_restarted.emit()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 786, in _handle_kernel_restarted
    super(ShellWidget, self)._handle_kernel_restarted()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 530, in _handle_kernel_restarted
    self.log.warning("kernel restarted")
Message: 'kernel restarted'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\application\container.py", line 286, in report_missing_dependencies
    QMessageBox.critical(
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 27, in poll
    super().poll()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\restarter.py", line 116, in poll
    self.kernel_manager.restart_kernel(now=True, newports=newports)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 510, in restart_kernel
    self.start_kernel(**self._launch_args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 342, in start_kernel
    self.post_start_kernel(**kw)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 62, in post_start_kernel
    self.kernel_restarted.emit()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 786, in _handle_kernel_restarted
    super(ShellWidget, self)._handle_kernel_restarted()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 530, in _handle_kernel_restarted
    self.log.warning("kernel restarted")
Message: 'kernel restarted'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\application\container.py", line 286, in report_missing_dependencies
    QMessageBox.critical(
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 27, in poll
    super().poll()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\restarter.py", line 116, in poll
    self.kernel_manager.restart_kernel(now=True, newports=newports)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 510, in restart_kernel
    self.start_kernel(**self._launch_args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 342, in start_kernel
    self.post_start_kernel(**kw)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 62, in post_start_kernel
    self.kernel_restarted.emit()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 786, in _handle_kernel_restarted
    super(ShellWidget, self)._handle_kernel_restarted()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 530, in _handle_kernel_restarted
    self.log.warning("kernel restarted")
Message: 'kernel restarted'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 27, in poll
    super().poll()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\restarter.py", line 116, in poll
    self.kernel_manager.restart_kernel(now=True, newports=newports)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 510, in restart_kernel
    self.start_kernel(**self._launch_args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\manager.py", line 342, in start_kernel
    self.post_start_kernel(**kw)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 62, in post_start_kernel
    self.kernel_restarted.emit()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 786, in _handle_kernel_restarted
    super(ShellWidget, self)._handle_kernel_restarted()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 530, in _handle_kernel_restarted
    self.log.warning("kernel restarted")
Message: 'kernel restarted'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\manager.py", line 27, in poll
    super().poll()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\jupyter_client\restarter.py", line 103, in poll
    self.log.warning("KernelRestarter: restart failed")
Message: 'KernelRestarter: restart failed'
Arguments: ()
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.011011123657227,)
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.01420021057129,)
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.014848470687866,)
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\widgets\browser.py", line 332, in contextMenuEvent
    self.context_menu.popup(event.globalPos())
AttributeError: 'NoneType' object has no attribute 'popup'
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\widgets\browser.py", line 332, in contextMenuEvent
    self.context_menu.popup(event.globalPos())
AttributeError: 'NoneType' object has no attribute 'popup'
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.001037120819092,)
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.005097150802612,)
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.004949808120728,)
--- Logging error ---
Traceback (most recent call last):
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\logging\__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
AttributeError: 'NoneType' object has no attribute 'write'
Call stack:
  File "C:\Users\jameswang\anaconda3\envs\sklearn\Scripts\spyder-script.py", line 10, in <module>
    sys.exit(main())
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\start.py", line 214, in main
    mainwindow.main(options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2293, in main
    mainwindow = create_window(app, splash, options, args)
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\spyder\app\mainwindow.py", line 2192, in create_window
    app.exec_()
  File "C:\Users\jameswang\anaconda3\envs\sklearn\lib\site-packages\qtconsole\frontend_widget.py", line 518, in _handle_kernel_died
    self.log.warning("kernel died: %s", since_last_heartbeat)
Message: 'kernel died: %s'
Arguments: (25.009049654006958,)
```

## Versions

* Spyder version: 5.0.0 
* Python version: 3.8.8
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.22.0 (OK)
jedi =0.17.2                  :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  22.3.0 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso =0.7.0                  :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.8.1 (OK)
pylint >=1.0                  :  2.7.4 (OK)
pyls >=0.36.2;<1.0.0          :  0.36.2 (OK)
pyls_black >=0.4.6            :  0.4.6 (OK)
pyls_spyder >=0.3.2           :  0.3.2 (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=0.5.7             :  1.0.2 (OK)
qtconsole >=5.0.3             :  5.0.3 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.8.3                 :  0.9.4 (OK)
setuptools >=39.0.0           :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                :  3.5.3 (OK)
spyder_kernels >=2.0.1,<2.1.0 :  2.0.1 (NOK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0      :  1.0.2 (OK)
zmq >=17                      :  20.0.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.19.2 (OK)
pandas >=1.1.1                :  1.2.3 (OK)
scipy >=0.17.0                :  1.6.2 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by dalthviz** [comment]
Hi @WANGCHIENCHIH thanks for the feedback! For the moment I think `spyder-notebook` is not yet compatible with Spyder 5. We will work on this in the future but for now if you need to use spyder-notebook use Spyder 4.

Pinging @ccordoba12 and @jitseniesen here just in case maybe they have a better perspective regarding spyder-notebook status

**Comment by ccordoba12** [comment]
This is a duplicate of issue #15112 and it'll be fixed in our next version (5.0.1), to be released soon.
